### DISCUSSION TITLE
Compilation error Source data/cert/x509_crt_bundle.bin not found, needed by target

### DISCUSSION THREAD

**Discussion by alexval2014** [HASwitchPlate/openHASP#327] [Q&A]
Hello. I had a problem when compiling in 0.6.3 in 0.6.2 there is no such problem.

*** [.pio/build/wt32-sc01_4MB/x509_crt_bundle.bin.txt.o] Source `data/cert/x509_crt_bundle.bin' not found, needed by target `.pio/build/wt32-sc01_4MB/x509_crt_bundle.bin.txt.o'.

`gitpod /workspace/openHASP $ platformio run
Processing wt32-sc01_4MB (board: esp32dev; framework: arduino; platform: https://github.com/tasmota/platform-espressif32)
--------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (2.0.3+sha.aacdc58) > Espressif ESP32 Dev Module
HARDWARE: ESP32S2 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 2.0.3 
 - tool-esptoolpy 1.30300.220323 (3.3.0) 
 - tool-mklittlefs 1.203.210628 (2.3) 
 - tool-mkspiffs 2.230.0 (2.30) 
 - toolchain-esp32ulp 1.22851.191205 (2.28.51) 
 - toolchain-xtensa-esp32s2 8.4.0+2021r2-patch3
Warning! Could not find file "x509_crt_bundle.bin"
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 50 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <AceButton> 1.9.2
|-- <ArduinoJson> 6.19.4
|-- <StreamUtils> 1.6.2
|   |-- <EEPROM> 2.0.0
|-- <PubSubClient> 2.8.0
|-- <lvgl> 7.11.1+sha.c539546
|-- <lv_lib_png> 0.0.0+20220428084136.sha.7ff6168
|   |-- <lvgl> 7.11.1+sha.c539546
|-- <lv_lib_bmp> 0.0.0+20220428084137.sha.b6ca85d
|   |-- <lvgl> 7.11.1+sha.c539546
|-- <lv_lib_gif> 0.0.0+20220428084137.sha.afdc2d6
|   |-- <lvgl> 7.11.1+sha.c539546
|-- <ConsoleInput> 0.0.3+sha.79180a9
|-- <TFT_eSPI> 2.4.51
|   |-- <SPIFFS> 2.0.0
|   |   |-- <FS> 2.0.0
|   |-- <FS> 2.0.0
|   |-- <LittleFS> 2.0.0
|   |   |-- <FS> 2.0.0
|   |-- <SPI> 2.0.0
|-- <FT6336U CTP Controller> 1.0.2+sha.d96d4be
|   |-- <Wire> 2.0.0
|-- <WiFi> 2.0.0
|-- <ArduinoLog>
|-- <Wire> 2.0.0
|-- <SPI> 2.0.0
|-- <Preferences> 2.0.0
|-- <EEPROM> 2.0.0
|-- <lv_fs_if> 0.0.0+20220202035348
|   |-- <lvgl> 7.11.1+sha.c539546
|-- <HTTPClient> 2.0.0
|   |-- <WiFi> 2.0.0
|   |-- <WiFiClientSecure> 2.0.0
|   |   |-- <WiFi> 2.0.0
|-- <CharStream>
|-- <StringStream>
|-- <lv_lib_freetype>
|   |-- <freetype> 2.10.4
|   |   |-- <lv_fs_if> 0.0.0+20220202035348
|   |   |   |-- <lvgl> 7.11.1+sha.c539546
|   |-- <lvgl> 7.11.1+sha.c539546
|   |-- <lv_fs_if> 0.0.0+20220202035348
|   |   |-- <lvgl> 7.11.1+sha.c539546
|-- <SPIFFS> 2.0.0
|   |-- <FS> 2.0.0
|-- <FS> 2.0.0
|-- <LittleFS> 2.0.0
|   |-- <FS> 2.0.0
|-- <lv_lib_qrcode>
|   |-- <lvgl> 7.11.1+sha.c539546
|-- <SimpleFTPServer>
|   |-- <FFat> 2.0.0
|   |   |-- <FS> 2.0.0
|   |-- <FS> 2.0.0
|   |-- <LittleFS> 2.0.0
|   |   |-- <FS> 2.0.0
|   |-- <Ethernet> 2.0.0
|   |   |-- <WiFi> 2.0.0
|   |-- <SD> 2.0.0
|   |   |-- <FS> 2.0.0
|   |   |-- <SPI> 2.0.0
|   |-- <SPI> 2.0.0
|   |-- <SPIFFS> 2.0.0
|   |   |-- <FS> 2.0.0
|   |-- <WiFi> 2.0.0
|-- <Update> 2.0.0
|-- <DNSServer> 2.0.0
|   |-- <WiFi> 2.0.0
|-- <WebServer> 2.0.0
|   |-- <WiFi> 2.0.0
|   |-- <FS> 2.0.0
|-- <ESPmDNS> 2.0.0
|   |-- <WiFi> 2.0.0
|-- <ArduinoOTA> 2.0.0
|   |-- <Update> 2.0.0
|   |-- <WiFi> 2.0.0
|   |-- <ESPmDNS> 2.0.0
|   |   |-- <WiFi> 2.0.0
|-- <HTTPUpdate> 2.0.0
|   |-- <HTTPClient> 2.0.0
|   |   |-- <WiFi> 2.0.0
|   |   |-- <WiFiClientSecure> 2.0.0
|   |   |   |-- <WiFi> 2.0.0
|   |-- <Update> 2.0.0
|   |-- <WiFi> 2.0.0
|-- <Ethernet> 2.0.0
|   |-- <WiFi> 2.0.0
Building in release mode
=====================================
Converting .pio/build/wt32-sc01_4MB/edit.htm.gz.txt.o
Converting .pio/build/wt32-sc01_4MB/style.css.gz.txt.o
Converting .pio/build/wt32-sc01_4MB/script.js.gz.txt.o
*** [.pio/build/wt32-sc01_4MB/x509_crt_bundle.bin.txt.o] Source `data/cert/x509_crt_bundle.bin' not found, needed by target `.pio/build/wt32-sc01_4MB/x509_crt_bundle.bin.txt.o'.
====================================== [FAILED] Took 4.57 seconds ======================================

Environment    Status    Duration
-------------  --------  ------------
wt32-sc01_4MB  FAILED    00:00:04.574
================================ 1 failed, 0 succeeded in 00:00:04.574 ================================
`

**Comment by fvanroie** [comment]
Pull the latest code and try again.

**Reply by alexval2014** [reply]
Thank you, this problem has already been solved.
